tests/suite/upstream_settings_test.go

```diff
@@ -0,0 +1,525 @@
+						return expectRequestToSucceed(baseTeaURL, address, "URI: /tea")
+					}).
+					WithTimeout(timeoutConfig.RequestTimeout).
+					WithPolling(1000 * time.Millisecond).
+					Should(Succeed())
+			})
+		})
+
+		Context("nginx directives", func() {
+			var conf *framework.Payload
+
+			BeforeAll(func() {
+				podNames, err := framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetTimeout)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(podNames).To(HaveLen(1))
+
+				ngfPodName := podNames[0]
+
+				conf, err = resourceManager.GetNginxConfig(ngfPodName, ngfNamespace)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			DescribeTable("are set properly for",
+				func(expCfgs []framework.ExpectedNginxField) {
+					for _, expCfg := range expCfgs {
+						Expect(framework.ValidateNginxFieldExists(conf, expCfg)).To(Succeed())
+					}
+				},
+				Entry("Coffee upstream", []framework.ExpectedNginxField{
+					{
+						Directive: "upstream",
+						Value:     "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "zone",
+						Value:     "uspolicy_coffee_80 512k",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive",
+						Value:     "100",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_requests",
+						Value:     "55",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_time",
+						Value:     "1m",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_timeout",
+						Value:     "5h",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+				}),
+				Entry("Tea upstream", []framework.ExpectedNginxField{
+					{
+						Directive: "zone",
+						Value:     "uspolicy_tea_80 128k",
+						Upstream:  "uspolicy_tea_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "upstream",
+						Value:     "uspolicy_tea_80",
+						File:      "http.conf",
+					},
+				}),
+			)
+		})
+	})
+
+	When("UpstreamSettingsPolicy targets a Service that does not exist", func() {
+		Specify("upstreamSettingsPolicy sets no condition", func() {
+			files := []string{"upstream-settings-policy/invalid-svc-usps.yaml"}
+
+			Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+			uspolicyNsName := types.NamespacedName{Name: "usps-target-not-found", Namespace: namespace}
+			ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetStatusTimeout)
+			defer cancel()
+
+			err := wait.PollUntilContextCancel(
+				ctx,
+				1000*time.Millisecond,
+				true, /* poll immediately */
+				func(ctx context.Context) (bool, error) {
+					return usPolicyHasNoAncestors(ctx, uspolicyNsName)
+				})
+
+			Expect(errors.Is(err, context.DeadlineExceeded)).To(BeTrue())
```

Comment:
Why this check?

Reply:
Because we consistently polling the function for an error. If we get any other error, the tests fail but it succeeds when error is context.DeadlineExceeded